md_docs/md_docs.go

```diff
@@ -0,0 +1,177 @@
+//Copyright 2015 Red Hat Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint
+package md_docs
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+// Print the arguments with 11 characters of padding
+func printArguments(buf *bytes.Buffer, cmd *cobra.Command, name string) error {
+	if len(cmd.Annotations) > 0 {
+		buf.WriteString("### Arguments\n\n```\n")
+		for k, v := range cmd.Annotations {
+			buf.WriteString(fmt.Sprintf("%-11v %s\n", k, v))
```

Comment:
Should we sort the arguments so they're the same order as the command line? We could re-use the same code for help text. I'm thinking something like:

```go

func PositionalArgs(cmd *cobra.command) []string {
    args := strings.Split(cmd.Use, " ")
    if len(args) <= 1 {
        return nil
    }
    return args[1:]
}

func FormatPositionalArg(cmd *cobra.Command, arg string) string {
    description, ok := cmd.Annotations[arg]
    if !ok {
        return ""
    }
    return fmt.Sprintf("%-11v %s\n", arg, description)
}

...

for _, arg := range PositionalArgs(cmd) {
    buf.WriteString(FormatPositionalArg(cmd, arg))
}
```
That would also enable the use of annotations for other purposes in the future.